segmentation/include/pcl/segmentation/impl/supervoxel_clustering.hpp

```diff
@@ -189,7 +189,7 @@ pcl::SupervoxelClustering<PointT>::prepareForSegmentation ()
   //double prep_start = timer_.getTime ();
   if ( (use_default_transform_behaviour_ && input_->isOrganized ())
        || (!use_default_transform_behaviour_ && use_single_camera_transform_))
-      adjacency_octree_->setTransformFunction (boost::bind (&SupervoxelClustering::transformFunction, this, _1));
+      adjacency_octree_->setTransformFunction ([this] (auto p1) { transformFunction (p1); });
```

Comment:
`transformFunction` modifies point in-place, so perhaps `[this] (auto& p1) ...`?

Reply:
I admit I'm a little unsure of the underlying behavior. I assumed I would not have to care if the `p1` needs to be a reference or not. The type resolution for `setTransformFunction` would handle it, but it might actually happen the other way around. 

Whatever you pass as an argument (usually not references)  will set the the `auto` deduction type. So the point will be copied and then a reference of that copy is passed down. This might be the reason why I had the segfaults in the changes above. I set the parameters on `auto` indiscriminately, didn't check if they needed the be refs.

Reply:
Hm, I am also not sure now. Either way, let's go with explicit signatures in such cases.